src/openms/include/OpenMS/ANALYSIS/ID/AhoCorasickDA.h

```diff
@@ -38,60 +38,45 @@
 #include <OpenMS/CONCEPT/Types.h>
 #include <vector>
 #include <unordered_map>
+#include <stdint.h>
 
 
 namespace OpenMS
 {
 
+  /**
+   * @brief Extended Aho-Corasick algorithm capable of matching ambiguous amino acids in the pattern (i.e. proteins).
+   * The trie of the Aho Corasick automaton is based on a compact double-array (DA) :
+   *
+   * Susumu  Yata  Masaki  Oono  Toru  Sumitomo  Kazuhiro  Morita,  Masao  Fuketa  and  Jun’ichiAoe.
+   * A  compact  static  double-array  keeping  character  codes.
+   * Information Processing and Managemnet : 237-247 (2007)
+   */
   class OPENMS_DLLAPI AhoCorasickDA
   {
 
   public:
 
-    /// constructor
+    /// constructor support DNA, RNA and Protein sequences (lower and capital letter) except ambiguous aa (B/J/X/Z)
     explicit AhoCorasickDA(const std::vector<String>& seq) : sequences_(seq)
     {
       construct_();
     }
 
     /// Destructor
-    virtual ~AhoCorasickDA() = default;
-
-
-
-    //needed for the construction
-    //contains informations about used nodes, used base values and empty elements
-    struct BuildInformation
-    {
-      UInt32 next        :31 ;    //next empty node in DA
-      UInt32 used_flag   :1  ;    //node is used
-      UInt32 prev        :31 ;    //previous empty node in the DA
-      UInt32 base_flag   :1  ;    //this value is used as base
-      UInt32 check           ;    //parent node TODO not necessary
-    };
-
-    struct BC
-    {
-      UInt32 base       :22 ;    // base value of DA
-      UInt32 lcheck     :8  ;    // arc label from parent node
-      UInt32 leaf_flag  :1  ;    // node is leaf
-      UInt32 term_flag  :1  ;
-    };
-
-    struct Spawn
-    {
-      Int32 node = 0;
-      UInt32 prot_pos = 0;
-      std::vector<UInt> a_acids{};
-    };
-
-
-
+    ~AhoCorasickDA() = default;
 
 
+    /**
+     * @brief 
+     * @param pos_in_protein position in the set protein
```

Comment:
to make it more clear, that this is an output parameter, you can use
` @param[out] pos_in_protein Position (zero-based) in the protein`
For better readability, make the first letter of 'Position' a capital letter and leave out the 'set', since at first, readers might confuse it with a noun, but here its used as an adjective